### PR TITLE
QR 3345 Services provides feedback when starting in background mode

### PR DESCRIPTION
**Pull request by git-jiby-me**
Enable service framework to provide more feedback and act more intelligent when starting up to handle initialisation issues, and not hanging in started state without actually starting.

TODO After merge
- [ ] Publish to npm

### CODE REVIEW

**Comment by ngalanis** [comment]
Done. Looks OK

File: lib/transports/file.js [hunk 1]
```diff
@@ -17,8 +18,13 @@ function getTransportConfig (params) {
         params.file = params.file;
     }
 
-    // make sure the folder exists, if we dont have permissions it will throw
-    mkdirp.sync(path.dirname(params.file));
+    try {
+        mkdirp.sync(path.dirname(params.file));
+        fs.accessSync(path.dirname(params.file), fs.W_OK);
+    } catch (err) {
+        throw new Error('Not enough permissions to write logs to ' + params.file);
```
Comment (jmasramon) [inline lib/transports/file.js]: Is it really this the only reason it can ever fail? What if it already exists? Or fs full?

Comment (git-jiby-me) [inline lib/transports/file.js]: @jmasramon we don't care if file exists, or if there is space in file system, we just care if we have permissions to write to the folder.

Comment (jmasramon) [inline lib/transports/file.js]: We do not care about much, do we?

**Pull request merged**